src/odemis/gui/comp/viewport.py

```diff
@@ -1065,7 +1066,175 @@ def _on_peak_method(self, state):
+        self.bottom_legend.Bind(wx.EVT_MOTION, self.on_hlegend_motion)
+
+    def onHRange(self, hrange):
+        self.bottom_legend.range = hrange
+        if self.canvas.has_data():
+            self.bottom_legend.lo_ellipsis = self.canvas.display_xrange[0] > self.canvas.data_xrange[0]
+            self.bottom_legend.hi_ellipsis = self.canvas.display_xrange[1] < self.canvas.data_xrange[1]
+        self.bottom_legend.Refresh()
+        self.canvas.set_ranges(self.hrange.value, self.vrange.value)
+
+    def onVRange(self, vrange):
+        self.left_legend.range = vrange
+        if self.canvas.has_data():
+            self.left_legend.lo_ellipsis = self.canvas.display_yrange[0] > self.canvas.data_yrange[0]
+            self.left_legend.hi_ellipsis = self.canvas.display_yrange[1] < self.canvas.data_yrange[1]
+        self.left_legend.Refresh()
+        self.canvas.set_ranges(self.hrange.value, self.vrange.value)
+
+    def on_hlegend_scroll(self, evt=None):
+
+        rot = evt.GetWheelRotation() / evt.GetWheelDelta()
+        span = self.hrange.value[1] - self.hrange.value[0]
+        zoom_centre = self.canvas.pos_x_to_val_x(evt.Position[0])
+        scale = 0.1
+        rng = self.hrange.value
+
+        # proportion of zoom_centre
+        prop = (zoom_centre - rng[0]) / span
+        
+        # zoom around the centre point
+        if rot < 0:
+            new_span = span * (1 + scale)
+        else:
+            new_span = span * (1 - scale)
+
+        lo = zoom_centre - prop * new_span
+        hi = lo + new_span
+
+        self.hrange.value = (lo, hi)
+        logging.debug("HRange: %s", (lo, hi))
+
+    def on_hlegend_drag_start(self, evt):
+        """ Start a dragging procedure """
+
+        # Ignore the click if we're already dragging
+        # self.set_dynamic_cursor(self._drag_cursor)
+
+        # Fixme: only go to drag mode if the mouse moves before a mouse up?
+        self._hdragging = True
+
+        pos = evt.Position
+        self.drag_init_pos = (pos[0], pos[1])
+        self._last_pos = (pos[0], pos[1])
+        self._last_time = time.time()
+
+        logging.debug("Drag started at %s", self.drag_init_pos)
+
+    def on_hlegend_drag_end(self, evt):
+        """ End the dragging procedure """
+
+        # If the canvas was being dragged
+        if self._hdragging:
+            self._hdragging = False
+
+        self.was_dragged = False
+
+    def on_hlegend_motion(self, evt):
+        """ Process mouse motion
+
+        Set the drag shift and refresh the image if dragging is enabled and the left mouse button is
+        down.
+
+        """
+        if self._hdragging:
+
+            v_pos = evt.Position
+            now = time.time()
+            speed = ((v_pos[0] - self._last_pos[0]) / (now - self._last_time),
+                     (v_pos[1] - self._last_pos[1]) / (now - self._last_time))
+
+            # if the mouse stopped, reset the init position
+            if abs(speed[0]) < 0.05:
```

Comment:
I don't understand the goal of resetting the drag_init_pos if the user is not fast enough. In general I think it's dangerous to rely  on speed, because some users might be slower that others.